ownership-chain/pallets/parachain-staking/src/traits.rs

```diff
@@ -46,25 +50,35 @@ impl OnNewRound for () {
 	}
 }
 
-/// Defines the behavior to payout the collator's reward.
-pub trait PayoutCollatorReward<Runtime: crate::Config> {
-	fn payout_collator_reward(
+/// Defines the behavior to payout the block producer reward.
+pub trait PayoutReward<Runtime: crate::Config, Balance> {
+	fn payout_with_computation_cost(
```

Comment:
hmm, any reason behind this? I thought this function's old name was fine.

Reply:
this is used for paying collators and `payout` is used for paying delegators, right?

Reply:
I have some reservations regarding the proposed renaming of certain functions:
- The function `payout_collator_rewards` is pivotal for distributing rewards to collators, indicating its purpose clearly through its name.
- Similarly, `deposit_into_existing` serves as a crucial abstraction layer, mirroring the functionality described in `pallet_balance::deposit_into_existing`. To enhance clarity and maintain consistency with the Polkadot SDK documentation, I suggest we annotate this trait with detailed comments. Specifically, these comments should elucidate that custom implementations are expected to replicate the behavior outlined in the Polkadot SDK for `deposit_into_existing`. 

For reference, the expected behavior as per the Polkadot SDK documentation ([https://paritytech.github.io/polkadot-sdk/master/frame/traits/trait.Currency.html#tymethod.deposit_into_existing](https://paritytech.github.io/polkadot-sdk/master/frame/traits/trait.Currency.html#tymethod.deposit_into_existing)) is as follows:

```
Transfer or Mint value to the free balance of the specified account (who).

If the account (who) does not exist, the operation is not carried out, and an error is returned instead.
```

Reply:
From the point of view of someone that wants to pay reward I think it doesn't matter which is the role of the rewarded account (collator, delegator or whatever). Actually I can pay collators without returning cost, the reason behind of having `payout_with_computation_cost` is because pallet staking uses it within a hook. But it might not be always the case

Reply:
> Transfer or Mint value to the free balance of the specified account (who).

it doesn't say `Transfer` anywhere
<img width="565" alt="Screenshot 2024-03-07 at 12 00 58" src="https://github.com/freeverseio/laos/assets/137785454/39ea67bb-ce4e-43f2-a13f-b4520335104b">

Reply:
maybe `deposit_into_existing` could be renamed to payment ... but let it generic cause it has more purposes that paying the delegators

Reply:
> > For reference, the expected behavior as per the Polkadot SDK documentation (https://paritytech.github.io/polkadot-sdk/master/frame/traits/trait.Currency.html#tymethod.deposit_into_existing) is as follows:
> > Transfer or Mint value to the free balance of the specified account (who).
> 
> it doesn't say `Transfer` anywhere <img alt="Screenshot 2024-03-07 at 12 00 58" width="565" src="https://private-user-images.githubusercontent.com/137785454/310853318-39ea67bb-ce4e-43f2-a13f-b4520335104b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk4MTc1MDEsIm5iZiI6MTcwOTgxNzIwMSwicGF0aCI6Ii8xMzc3ODU0NTQvMzEwODUzMzE4LTM5ZWE2N2JiLWNlNGUtNDNmMi1hMTNmLWI0NTIwMzM1MTA0Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwN1QxMzEzMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZTlhMzE2YmJiZTNkZjRlNmI2MTA2MzRmYWVmYjY3ZjZlZTM1N2UwMmM1NWI2ZDkzOThkYWViMTMwNDA2MGI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.p_SjU7vc7ZntjpVGEGF39Is32zQEovk3vrPaknyoNHw">

Totally right .. I changed it to propose the comments for the trait. My bad.